asllib/Typing.ml

```diff
@@ -1740,28 +1753,31 @@ module Annotate (C : ANNOTATE_CONFIG) = struct
     (* Begin SFor *)
     | S_For (id, e1, dir, e2, s') ->
         let t1, e1' = annotate_expr env e1 and t2, e2' = annotate_expr env e2 in
-        let+ () = check_structure_integer s' env t1 in
-        let+ () = check_structure_integer s' env t2 in
+        let struct1 = Types.get_structure env t1 |> Types.to_well_constrained
```

Comment:
Since you use the "Types.get_structure env t1 |> Types.to_well_constrained" combination more than a few times, perhaps add a new function to reuse? Perhaps something like "get_well_constrained_structure"?
Actually, would it work if Typing.get_structure already used to_well_constrained internally?

Reply:
Okay, let's go with `get_well_constrained_structure` then.